gh-134976: document the exception type that can be raised by `s[i]` (#134977)
python/cpython: Doc/library/stdtypes.rst

   without copying any data and with the returned index being relative to
   the start of the sequence rather than the start of the slice.

(9)

>    If the sequence is empty or *i* is outside the sequence range, :exc:`IndexError`
>    will be raised.

Reviewer: ```suggestion
   An :exc:`IndexError` is raised if *i* is outside the sequence range.
```

If the sequence is empty, i will always be outside the sequence range so it's fine. OTOH, I'm not sure if it's sufficiently clear that *i* here means *i* after it's been converted to `len(s) + i` if `i < 0`.

Author: Thanks for suggestions!

Author: > I'm not sure if it's sufficiently clear that i here means i after it's been converted to len(s) + i if i < 0.

I think users can understand it. (3) said.
